add episodes on estimate superspreading and simulate chains (trial 04) (#29)
epiverse-trace/tutorials-middle: episodes/superspreading-simulate.Rmd

---
title: 'Simulate transmission chains'
teaching: 10
exercises: 2
---

:::::::::::::::::::::::::::::::::::::: questions 

> - How to simulate transmission chains?

Reviewer: ```suggestion
- How can we simulate transmission chains based on infection characteristics?
```

::::::::::::::::::::::::::::::::::::::::::::::::

::::::::::::::::::::::::::::::::::::: objectives

> - Create a short term projection using branching process with `{epichains}`.

Reviewer: ```suggestion
- Create a short term projection using a branching process with `{epichains}`.
```

::::::::::::::::::::::::::::::::::::::::::::::::

::::::::::::::::::::::::::::::::::::: prereq

## Prerequisites

Learners should familiarise themselves with following concept dependencies before working through this tutorial: 

> **Statistics**: probability distributions. 

Reviewer: ```suggestion
**Statistics**: Common probability distributions, including Poisson and negative binomial.
```

> **Epidemic theory**: Reproduction number.

Reviewer: ```suggestion
**Epidemic theory**: The reproduction number, $R$.
```

---
title: 'Simulate transmission chains'
teaching: 30
exercises: 2
---

:::::::::::::::::::::::::::::::::::::: questions 

- How to simulate transmission chains?

::::::::::::::::::::::::::::::::::::::::::::::::

::::::::::::::::::::::::::::::::::::: objectives

- Create a short term projection using branching process with `{epichains}`.

::::::::::::::::::::::::::::::::::::::::::::::::

::::::::::::::::::::::::::::::::::::: prereq

## Prerequisites

Learners should familiarise themselves with the following concept dependencies before working through this tutorial: 

**Statistics**: probability distributions. 

**Epidemic theory**: Reproduction number.

:::::::::::::::::::::::::::::::::

## Introduction

<!-- what we know -->

> Transmission heterogeneity affects epidemic establishment and the ease of control ([Cori et al., 2017](https://royalsocietypublishing.org/doi/10.1098/rstb.2016.0371)). 

Author: homologized in https://github.com/epiverse-trace/tutorials-middle/pull/29/commits/9d846e2e2aebf8b8a67d97a96b478eb1144b9e01

+ Conversely, heterogeneity in transmission may provide opportunities for *targeted interventions* if the individuals who contribute more to transmission (due to biological, environmental or behavioural factors) share socio-demographic or geographical characteristics that can be defined.

<!-- what we dont know -->

However, how can we understand the potential of a disease epidemic early from the mean reproduction number $R$ and the dispersion $k$ of an offspring distribution?

```{r,echo=FALSE,warning=FALSE,message=FALSE,fig.cap="Observed number of cumulative cases from the Middle East respiratory syndrome (MERS) outbreak in South Korea, 2015, alongside with simulated transmission chains given a known offspring distribution."}
library(epichains)
library(epiparameter)
library(tidyverse)
library(outbreaks)

mers_cumcases <- mers_korea_2015$linelist %>%
  # incidence2 workflow
  incidence2::incidence(date_index = "dt_onset") %>%
  incidence2::complete_dates() %>%
  # wrangling using {dplyr}
  mutate(count_cumsum = cumsum(count)) %>%
  rownames_to_column(var = "day") %>%
  mutate(day = as.numeric(day))

# offspring distribution

> mers_offspring <- c(mean = 0.60, dispersion = 0.02)

Author: the k=0.02 is extreme, indeed, but is the result we get from the previous episode. Should we add this paper and others to briefly share more context about the estimate? the dataset came from https://www.eurosurveillance.org/content/10.2807/1560-7917.ES2013.18.39.20596

> mers_offspring <- c(mean = 0.60, dispersion = 0.02)

Author: added as issue in #66

::::::::::::::::::::::::::::::::

Let's use the `{epiparameter}` package to access and use the available serial interval for MERS disease:

```{r,message=FALSE,warning=FALSE}
serial_inverval <- epidist_db(
  disease = "mers",
  epi_dist = "serial",
  single_epidist = TRUE
)

plot(serial_inverval, day_range = 0:25)
```

```{r,echo=FALSE}
serial_summary <- serial_inverval$summary_stats %>%
  as_tibble() %>%
  select(mean, sd) %>%
  distinct(.keep_all = TRUE)
```

The serial interval for MERS has a mean of `r serial_summary$mean` days and a standard deviation of `r serial_summary$sd` days.

:::::::::::::::::::::::::::: callout

### generation time for epichains

In `{epichains}`, we need to specify the generation time as a function that generates random numbers. Using `{epiparameter}` has the advantage of using the distribution function `epiparameter::generate()` for this input. This will look like this:

```r

> function(x) generate(x = serial_inverval, times = x)

Author: replied in https://github.com/epiverse-trace/tutorials-middle/pull/29#discussion_r1566556054

```

This interface is similar to the one `{cfr}` uses to link with `{epiparameter}`. Read the [work with delay distributions](https://epiverse-trace.github.io/cfr/articles/delay_distributions.html) vignette for further context.

::::::::::::::::::::::::::::

## Simulate a single chain

Now we are prepared to use the `simulate_chains()` function from `{epichains}` to create **one** transmission chain:

```{r,message=FALSE,warning=FALSE,eval=FALSE}
simulate_chains(
  # simulation controls
  index_cases = 5,
  statistic = "size",
  # offspring
  offspring_dist = rnbinom,
  mu = mers_offspring["mean"],
  size = mers_offspring["dispersion"],
  # generation

>   generation_time = function(x) generate(x = serial_inverval, times = x)

Author: It is not, currently. This works in a similar fashion as in {cfr} ([explained in vignette](https://epiverse-trace.github.io/cfr/articles/delay_distributions.html#links-to-epiparameter)), but for {simulist}, AFAIK. I'll build a reprex to share as an issue to cross-communicate between {cfr}, {epichains} and {simulist} for API consistency

>   generation_time = function(x) generate(x = serial_inverval, times = x)

Author: issue filled in https://github.com/epiverse-trace/simulist/issues/107

)
```

`simulate_chains()` requires at least three sets of arguments:

- simulation controls,
- offspring distribution, and
- generation time.

In the lines above, we described how to specify the offspring distribution and generation time. The **simulation controls** include at least two arguments:

- `index_case`, which defines the number of index cases to simulate transmission chains for and
- `statistic`, which defines a chain statistic to track (either `"size"` or `"length"`) as the stopping criteria for each chain being simulated.

::::::::::::::::::::::::::: callout

### Stopping criteria

> This is an exclusive feature of `{epichains}`. Branching process simulations end when they have gone extinct. No more offspring are being produced because of some stopping criterion.

Author: fixed in https://github.com/epiverse-trace/tutorials-middle/commit/0e054ad60a09e9c5e1137c37cbeaaa62fe0e23ba

:::::::::::::::::::::::::::

> The `simulate_chains()` output creates a `<epichains>` class object.

Reviewer: ```suggestion
The `simulate_chains()` output creates a `<epichains>` class object, which we can then analyse further in R.
```

```{r}
# Set seed for random number generator
set.seed(33)
# Number of simulation runs
number_chains <- 1000
# Number of initial cases
initial_cases <- 1
```

`number_chains` and `initial_cases` are conveniently stored in objects to facilitate downstream reuse in the workflow.

:::::::::::::::::::::::::::::: checklist

### Iteration using purrr

[Iteration](https://r4ds.hadley.nz/iteration.html) aims to perform the same action on different objects repeatedly.

Learn how to use the core `{purrr}` functions like `map()` from the YouTube tutorial on [How to purrr](https://www.youtube.com/watch?v=nd-Y8b22YaQ) by Equitable Equations.

Or, if you previously used the `*apply` family of functions, visit the package vignette on [purrr base R](https://purrr.tidyverse.org/articles/base.html), which shares key differences, direct translations, and examples.

::::::::::::::::::::::::::::::

To get multiple chains, we must apply the `simulate_chains()` function to each chain defined by a sequence of numbers from 1 to `number_chains`.

::::::::::::::::::::::::::::::: callout

### purrr and epichains

> First, let's sketch how we use `purrr::map()` with `epichains::simulate_chains()`. The `map()` function requires two arguments:

Reviewer: Worth mentioning (or reminding) that `::` is just calling a function from a specific package?

Author: testing to have this right after loading packages in all episodes https://github.com/epiverse-trace/tutorials-middle/commit/b00d13bb28c205086093e68a8a7a52d49bb11d8d

> - `.x`, with a vector of numbers, and

Author: added in https://github.com/epiverse-trace/tutorials-middle/commit/171028626419b00712a2f70699b319900431875c

- `.f`, a function to iterate to each vector value.

```r
map(
  # vector of numbers (chain IDs)
  .x = seq_len(number_chains),
  # function to iterate to each chain ID number
  .f = function(sim) {
    simulate_chains(...) %>%
      # creates a column with the chain ID number
      mutate(chain_id = sim)
  }
) %>%
  # combine list outputs (for each chain ID) into a single data frame
  list_rbind()
```

The `sim` element is placed to register the iteration number (**chain ID**) as a new column in the `<epichains>` output. The `purrr::list_rbind()` function aims to combine all the list outputs from `map()`.

:::::::::::::::::::::::::::::::

> Now, we are prepared to use `map()` to iterate `simulate_chains()` in a vector from 1 to `r number_chains`:

Reviewer: ```suggestion
Now, we are prepared to use `map()` to repeatedly simulate from `simulate_chains()` and store in a vector from 1 to `number_chains`:
```

        offspring_dist = rnbinom,
        mu = mers_offspring["mean"],
        size = mers_offspring["dispersion"],
        # generation
        generation_time = function(x) generate(x = serial_inverval, times = x)
      ) %>%
        # creates a column with the chain ID number
        mutate(chain_id = sim)
    }
  ) %>%
  # combine list outputs (for each chain ID) into a single data frame
  list_rbind()
```

```{r,echo=FALSE,eval=FALSE}
# view infectee number per simulation
simulated_chains_map %>%
  as_tibble() %>%
  count(chain_id, sort = TRUE)
```

::::::::::::::::::::::::::::::::: discussion

### Read the epichains output

```{r,echo=FALSE}
#### get simulated chain with more infections --------------------------------

chains_subgroup <- simulated_chains_map %>%
  as_tibble() %>%
  group_by(chain_id) %>%
  distinct(infector_id) %>%
  count(chain_id) %>%
  filter(n < 5, n >= 3) %>%
  pull(chain_id)

chain_to_observe <- simulated_chains_map %>%
  as_tibble() %>%
  count(chain_id, sort = TRUE) %>%
  filter(
    magrittr::is_in(
      chain_id, chains_subgroup
    )
  ) %>%
  filter(n < 10) %>%
  filter(n == max(n)) %>%
  slice_min(chain_id) %>%
  pull(chain_id)
```

> To explore the output format of the `<epichains>` class object of name `simulated_chains_map`, let's observe closely the simulated `chain_id` number `r chain_to_observe`. 

Reviewer: ```suggestion
To explore the output format of the `<epichains>` class object of name `simulated_chains_map`, let's look at the simulated `chain_id` number `r chain_to_observe`. 
```
Is `r chain_to_observe` a typo? (See also above edit to `r number_chains`)?

Author: > Is `r chain_to_observe` a typo? (See also above edit to `r number_chains`)?

fixed both in https://github.com/epiverse-trace/tutorials-middle/pull/29/commits/01d2f3b0abcdcf8b8fd73833e73f901cf725bd56 and https://github.com/epiverse-trace/tutorials-middle/pull/29/commits/01d2f3b0abcdcf8b8fd73833e73f901cf725bd56

Author: > Is `r chain_to_observe` a typo?

it is not. we use `r object` to print `object` in the rendered text.

:::::::::::::::::::::::::::::::::

::::::::::::::::::::::::: solution

> ### the epichains object

Reviewer: ```suggestion
### The epichains object
```

Let's use `dplyr::filter()` for this:

```{r}
#### get epichain summary ----------------------------------------------------

simulated_chains_map %>%
  filter(chain_id == chain_to_observe)
```

Key elements exclusive from this output are in the **footer**:

```output
Number of infectors (known): 3
Number of generations: 3
```

> The simulated `chain_id` number `r chain_to_observe` has three known infectors and three generations. These numbers are more visible when reading the `<epichains>` objects as a data frame.

Reviewer: See above about `r chain_to_observe` possible typo (will assume any further down in doc are either fine or can be fixed!)

:::::::::::::::::::::::::

::::::::::::::::::::::::: solution

> ### the epichains data frame

Reviewer: Check capitalisation consistency?

Author: fixed in https://github.com/epiverse-trace/tutorials-middle/pull/29/commits/fc53f05fc383e4d427735e95589a611a9efcd936

chains_extinct <- sim_chains_max %>%
  arrange(desc(day_max)) %>%
  filter(day_max > 0) %>%
  summarise(
    extinct_duration_median = median(day_max),
    extinct_size_median = median(cases_total)
  )

chains_null <- sim_chains_max %>%
  arrange(desc(cases_total)) %>%
  filter(day_max < 1) %>%
  nrow()
```

Although most introductions of `r initial_cases` index case do not generate secondary cases (N = `r chains_null`) or most outbreaks rapidly become extinct (median duration of `r chains_extinct$extinct_duration_median` and median size of `r chains_extinct$extinct_size_median`), only `r threshhold_summary$chains_theshold` epidemic trajectories among `r threshhold_summary$chains_number` simulations (`r threshhold_summary$chains_percentage`%) can reach to more than 100 infected cases. This finding is particularly remarkable because the reproduction number $R$ is less than 1 (offspring distribution mean of `r mers_offspring[["mean"]]`), but, given an offspring distribution dispersion parameter of `r mers_offspring[["dispersion"]]`, it shows the potential for explosive outbreaks of MERS disease.

::::::::::::::::::::::::::::::::::: spoiler

### Observed cases vs simulated chains

> Let's overlap the cumulative number of observed cases using the linelist object from the `mers_korea_2015` dataset of the `{outbreaks}` R package. To prepare the dataset we use `{incidence2}` to convert the linelist to an `<incidence2>` object, complete the missing dates of the time series with `complete_dates()`

Reviewer: ```suggestion
Let's overlap the cumulative number of observed cases using the linelist object from the `mers_korea_2015` dataset of the `{outbreaks}` R package. To prepare the dataset so we can plot daily total cases over time, we use `{incidence2}` to convert the linelist to an `<incidence2>` object, complete the missing dates of the time series with `complete_dates()`
```

```{r,warning=FALSE,message=FALSE}
library(outbreaks)

mers_cumcases <- mers_korea_2015$linelist %>%
  # incidence2 workflow
  incidence2::incidence(date_index = "dt_onset") %>%
  incidence2::complete_dates() %>%
  # wrangling using {dplyr}
  mutate(count_cumsum = cumsum(count)) %>%
  rownames_to_column(var = "day") %>%
  mutate(day = as.numeric(day))
```

Use `plot()` to make an incidence plot:

```{r}
# plot the incidence2 object
plot(mers_cumcases)
```

:::::::::::::::::::::::::::::::::::

> When plotting the observed number of cumulative cases from the Middle East respiratory syndrome (MERS) outbreak in South Korea in 2015 alongside the previously simulated chains, we identify that the observed cases followed a trajectory that resembled a simulated explosive outbreak.

Reviewer: ```suggestion
When plotting the observed number of cumulative cases from the Middle East respiratory syndrome (MERS) outbreak in South Korea in 2015 alongside the previously simulated chains, we see that the observed cases followed a trajectory that is consistent with the simulated explosive outbreak dynamics (which makes sense, given the simulation uses parameters based on this specific outbreak).
```

```{r,echo=FALSE}
# Observed cases vs Simulated transmission chains
mers_cumcases_type <-
  mers_cumcases %>%
  mutate(type = "Observed")
simulated_chains_day_type <-
  simulated_chains_day %>%
  mutate(type = "Simulated")

ggplot() +
  geom_line(
    data = simulated_chains_day_type,
    mapping = aes(
      x = day,
      y = cases_cumsum,
      group = chain_id,
      color = type
    )
  ) +
  geom_hline(aes(yintercept = 100), lty = 2) +
  geom_line(
    data = mers_cumcases_type,
    mapping = aes(
      x = day,
      y = count_cumsum,
      color = type
    ),
    linewidth = 1.5
  ) +
  labs(
    x = "Day since first report",
    y = "Cumulative cases",
    color = "Type"
  )
```

> When we increase the dispersion parameter from $k = 0.01$ to $k = \infty$, given a fixed reproduction number $R = 1.5$, the proportion of simulated outbreaks that reached the 100-case threshold increases.

Reviewer: ```suggestion
When we increase the dispersion parameter from $k = 0.01$ to $k = \infty$ - and hence reduce individual-level variation in transmission - and assume a fixed reproduction number $R = 1.5$, the proportion of simulated outbreaks that reached the 100-case threshold increases. This is because the simulated outbreaks now have more of a consistent, clockwise dynamic, rather than the high level of variability seen previously.
```

![**Growth of simulated outbreaks with R = 1.5 and one initial case, conditional on non-extinction.** Boxes show the median and interquartile range (IQR) of the first disease generation with 100 cases; whiskers show the most extreme values within 1.5 × IQR of the boxes, and crosses show outliers. Percentages show the proportion of 10,000 simulated outbreaks that reached the 100-case threshold ([Lloyd-Smith et al., 2005](https://www.nature.com/articles/nature04153)).](fig/see-nature04153_Fig2-c.jpg)

:::::::::::::::::::::: testimonial

### Early spread projections

> In the epidemic's initial phase, you can use `{epichains}` to apply a branching process model to project the number of future cases. Validation analysis showed that this is a rapid and low-detail model with forecast prediction intervals that are too certain ([Pearson et al., 2020](https://www.eurosurveillance.org/content/10.2807/1560-7917.ES.2020.25.18.2000543#validationofthemodel-1)).

Reviewer: ```suggestion
In the epidemic's initial phase, you can use `{epichains}` to apply a branching process model to project the number of future cases. Even though the model accounts for randomness in transmission and variation in the number of secondary cases, there may be additional local features we have not considered. Analysis of early forecasts made for COVID in different countries using this model structure found that predictions were often overconfident ([Pearson et al., 2020](https://www.eurosurveillance.org/content/10.2807/1560-7917.ES.2020.25.18.2000543#validationofthemodel-1)). This is likely because the real-time model did not include all the changes in the offspring distribution that were happening at the local level as a result of behaviour change and control measures. You can read more about the importance of local context in COVID-19 models in [Eggo et al. (2020)](https://www.nature.com/articles/s43588-020-00014-7).
```
Have tried to broaden implications a bit for users.

> We invite you to read the vignette on [Projecting infectious disease incidence: a COVID-19 example](https://epiverse-trace.github.io/epichains/articles/projecting_incidence.html)!

Reviewer: ```suggestion
We invite you to read the vignette on [Projecting infectious disease incidence: a COVID-19 example](https://epiverse-trace.github.io/epichains/articles/projecting_incidence.html)! for more on making predictions using `{epichains}`.
```

::::::::::::::::::::::

## Challenges

:::::::::::::::::::::::::: challenge

### Ebola's large outbreak potential

> Evaluate the potential of Ebola to generate explosive outbreaks. Use the appropriate package to access delay data from previous outbreaks.

Reviewer: ```suggestion
Evaluate the potential for a new Ebola case to generate an explosive large outbreak. Use the appropriate package to access delay data from previous outbreaks.
```

### From a distribution of secondary cases

[Christian Althaus, 2015](https://www.thelancet.com/journals/laninf/article/PIIS1473-3099(15)70135-0/fulltext) reused data published by [Faye et al., 2015 (Figure 2)](https://www.thelancet.com/journals/laninf/article/PIIS1473-3099(14)71075-8/fulltext#gr2) on the transmission tree on Ebola virus disease in Conakry, Guinea, 2014.

Using the data under the **hint** tab, estimate the offspring distribution from the distribution of secondary cases. Then estimate the large outbreak potential from this data.

::::::::::: hint

Code with the transmission tree data written by [Christian Althaus, 2015](https://www.thelancet.com/journals/laninf/article/PIIS1473-3099(15)70135-0/fulltext):

```{r,message=FALSE,warning=FALSE}
# Number of individuals in the trees
n <- 152
# Number of secondary cases for all individuals
c1 <- c(1, 2, 2, 5, 14, 1, 4, 4, 1, 3, 3, 8, 2, 1, 1,
        4, 9, 9, 1, 1, 17, 2, 1, 1, 1, 4, 3, 3, 4, 2,
        5, 1, 2, 2, 1, 9, 1, 3, 1, 2, 1, 1, 2)
c0 <- c(c1, rep(0, n - length(c1)))

c0 %>%
  enframe() %>%
  ggplot(aes(value)) +
  geom_histogram()
```

:::::::::::

::::::::::: solution

<!-- [pending to complete] -->

> Remarkable, even with R0 less than 1

Reviewer: ```suggestion
Remarkably, even with R0 less than 1
```
Also, should this be one paragraph, rather than having a line break?

Author: completed and fixed in https://github.com/epiverse-trace/tutorials-middle/commit/60ceb9f801222656f8bc3ecaf7a72132096e3256

with potentially explosive outbreaks. The observed variation in individual infectiousness in Ebola
means that although the probability of extinction is high, new index cases also have the potential for explosive regrowth of the epidemic.

:::::::::::

:::::::::::::::::::::::::::::

:::::::::::::::::::: challenge

### Probability to contain an epidemic

> Calculate probabilities for Ebola of a large outbreak and to contain an epidemic. Use the appropriate package to access delay data from previous outbreaks.

Reviewer: ```suggestion
Calculate the probability a new Ebola case will lead to a large outbreak in the absence of control measures. Use the appropriate package to access delay data from previous outbreaks.
```
Not sure what containment means here – as control measures would reduce $R$ in reality? Have suggested an edit

Author: removed but added as hidden text in https://github.com/epiverse-trace/tutorials-middle/commit/60ceb9f801222656f8bc3ecaf7a72132096e3256

:::::::::::::::::: hint

With `{epiparameter}` you can access and use delay distributions from previous outbreaks.

:::::::::::::::::::::

::::::::::::::::::: hint

### Epidemic Risk assessment accounting for superspreading

With `{superspreading}`, you can get numerical solutions to processes that `{epichains}` solve using branching processes. We invite you to read the `{superspreading}` vignette on [Epidemic risk](https://epiverse-trace.github.io/superspreading/articles/epidemic_risk.html) and respond to:

- What is the probability of a pathogen causing a large outbreak?
- What is the probability of an outbreak becoming extinct?
- What is the probability of containing the ongoing outbreak?

Check how these estimates vary non-linearly with respect to the mean reproduction number $R$ and dispersion $k$ of a given disease.

::::::::::::::::::

::::::::::::::::::::: hint

Here is an example with the MERS offspring distribution estimates:

```{r,message=FALSE,warning=FALSE}
library(superspreading)
```

```{r}
# estimate probability to contain
probability_contain(

>   R = mers_offspring["mean"],

Reviewer: Do you mean Ebola (see comment above)? Although MERS also fine as a case study

Author: removed but placed hidden for further evaluation in https://github.com/epiverse-trace/tutorials-middle/commit/60ceb9f801222656f8bc3ecaf7a72132096e3256

  k = mers_offspring["dispersion"],
  num_init_infect = 1,
  case_threshold = 100
)

# Estimate the probability of a large outbreak
# with 5 independent imported cases
probability_epidemic(
  R = mers_offspring["mean"],
  k = mers_offspring["dispersion"],
  num_init_infect = 5
)
```

::::::::::::::::::

:::::::::::::::::: solution

```{r}

> #' get ebola params

Reviewer: To come?

Author: completed and fixed in https://github.com/epiverse-trace/tutorials-middle/commit/60ceb9f801222656f8bc3ecaf7a72132096e3256

---
title: 'Simulate transmission chains'
teaching: 30
exercises: 2
---

:::::::::::::::::::::::::::::::::::::: questions 

- How can we simulate transmission chains based on infection characteristics?

::::::::::::::::::::::::::::::::::::::::::::::::

::::::::::::::::::::::::::::::::::::: objectives

- Create a short term projection using a branching process with `{epichains}`.

::::::::::::::::::::::::::::::::::::::::::::::::

::::::::::::::::::::::::::::::::::::: prereq

## Prerequisites

Learners should familiarise themselves with the following concept dependencies before working through this tutorial: 

**Statistics**: Common probability distributions, including Poisson and negative binomial.

**Epidemic theory**: The reproduction number, $R$.

:::::::::::::::::::::::::::::::::

## Introduction

<!-- what we know -->

> Individual variation in transmission can affects both the potential for an epidemic to establish in a population and the ease of control ([Cori et al., 2017](https://royalsocietypublishing.org/doi/10.1098/rstb.2016.0371)). 

Reviewer 2: ```suggestion
Individual variation in transmission can affect both the potential for an epidemic to establish in a population and the ease of control ([Cori et al., 2017](https://royalsocietypublishing.org/doi/10.1098/rstb.2016.0371)). 
```

<!-- what we dont know -->

How can we quantify the potential of a new infection to cause a large outbreak based on its reproduction number $R$ and the dispersion $k$ of its offspring distribution?

```{r,echo=FALSE,warning=FALSE,message=FALSE,fig.cap="Observed number of cumulative cases from the Middle East respiratory syndrome (MERS) outbreak in South Korea, 2015, alongside with simulated transmission chains assuming an offspring distribution with $R=0.6$ and $k=0.02$."}
library(epichains)
library(epiparameter)
library(tidyverse)
library(outbreaks)

mers_cumcases <- mers_korea_2015$linelist %>%
  # incidence2 workflow
  incidence2::incidence(date_index = "dt_onset") %>%
  incidence2::complete_dates() %>%
  # wrangling using {dplyr}
  mutate(count_cumsum = cumsum(count)) %>%
  rownames_to_column(var = "day") %>%
  mutate(day = as.numeric(day))

# offspring distribution
mers_offspring <- c(mean = 0.60, dispersion = 0.02)

# generation time

> serial_inverval <- epidist_db(

Reviewer 2: ```suggestion
serial_interval <- epidist_db(
```

Author: thanks! this propagated across the episode hehe

Author: fixed in https://github.com/epiverse-trace/tutorials-middle/commit/171028626419b00712a2f70699b319900431875c

  disease = "mers",
  epi_dist = "serial",
  single_epidist = TRUE
)

# Set seed for random number generator
set.seed(33)
# Number of simulation runs
number_chains <- 1000
# Number of initial cases
initial_cases <- 1

simulated_chains_map <-
  # iterate one function across multiple numbers (chain IDs)
  map(
    # vector of numbers (chain IDs)
    .x = seq_len(number_chains),
    # function to iterate to each chain ID number
    .f = function(sim) {
      simulate_chains(
        # simulation controls
        index_cases = initial_cases,
        statistic = "size",
        # offspring
        offspring_dist = rnbinom,
        mu = mers_offspring["mean"],
        size = mers_offspring["dispersion"],
        # generation

>         generation_time = function(x) generate(x = serial_inverval, times = x)

Reviewer 2: ```suggestion
        generation_time = function(x) generate(x = serial_interval, times = x)
```

Author: fixed in https://github.com/epiverse-trace/tutorials-middle/commit/171028626419b00712a2f70699b319900431875c
